### PR TITLE
Add Ruby as a possible language

### PR DESCRIPTION
**Pull request by aibaars**
### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: src/codeql.ts [hunk 1]
```diff
@@ -654,6 +663,19 @@ function getCodeQLForCmd(cmd: string): CodeQL {
...
+      let output = "";
+      await new toolrunner.ToolRunner(cmd, codeqlArgs, {
+        listeners: {
+          stdout: (data: Buffer) => {
+            output += data.toString();
+          },
+        },
+      }).exec();
+
+      return JSON.parse(output);
```
Comment (aeisenberg) [inline src/codeql.ts:677]: Catch malformed json errors and rethrow with a meaningful message.

```suggestion
      try {
        return JSON.parse(output);
      } catch (e) {
        throw new Error("Something meaningful.);
      }
```

Comment (aibaars) [reply src/codeql.ts:677]: Wouldn't that suppress the likely interesting error message of `e` ? In Java I'd write `new Error("Unexpected output from codeql resolve languages", e)` . What would be the equivalent in Javascript?

Comment (robertbrignull) [reply src/codeql.ts:677]: I think just construct the message as a string. So like `new Error("Unexpected output from codeql resolve languages: " + e.message)` and maybe some type checks to make sure that `e.message` exists.

Comment (aibaars) [reply src/codeql.ts]: I opted to just write `Unexpected output from codeql resolve languages: ${e}` . This should include the original error's type and side-steps the problem of a potentially empty message string.

**Review by aeisenberg — APPROVED** [review]
We're populating the changelog now. Since this is a public-facing feature, could you add an entry. And then it's good to merge. Thanks!

**Comment by aibaars** [comment]
> We're populating the changelog now. Since this is a public-facing feature, could you add an entry. And then it's good to merge. Thanks!

There shouldn't be an entry. This is just for internal tests. CodeQL for Ruby is not available yet.

**Pull request merged**